grouper/initialization.py

```diff
@@ -0,0 +1,25 @@
+"""Setup functions common to all Grouper UIs."""
+
+from typing import TYPE_CHECKING
+
+from grouper.repositories.factory import RepositoryFactory
+from grouper.services.factory import ServiceFactory
+from grouper.usecases.factory import UseCaseFactory
+
+if TYPE_CHECKING:
+    from grouper.graph import GroupGraph
+    from grouper.models.base.session import Session
+    from grouper.settings import Settings
+    from typing import Optional
+
+
+def create_usecase_factory(settings, session=None, graph=None):
+    # type: (Settings, Optional[Session], Optional[GroupGraph]) -> UseCaseFactory
+    """Create a UseCaseFactory, with optional injection of a Session and GroupGraph.
+
+    Session and GroupGraph injection are supported primarily for tests.  If not injected, they will
```

Comment:
Looks like we still inject these in production code. When that changes, I'd suggest removing the optional args from this stack (i.e. this method and the constructor). One option worth considering is to pull the session and graph into their own factory/provider and to have two implementations for this, e.g. a DynamicSessionFactory and StaticSessionFactory with the latter being only used in tests. Yay dependency injection! =)

Reply:
Yup, that sounds like a great idea, and yeah, I only didn't push this farther because right now we're very inconsistent about when we create sessions and graphs and I didn't want to mess too much with the existing code.